examples/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientTls.java

```diff
@@ -0,0 +1,128 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldtls;
+
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.examples.helloworld.HelloWorldServer;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+
+import javax.net.ssl.SSLException;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A simple client that requests a greeting from the {@link HelloWorldServer} with TLS.
+ */
+public class HelloWorldClientTls {
+  private static final Logger logger = Logger.getLogger(HelloWorldClientTls.class.getName());
+
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  private static SslContext buildSslContext(String trustCertCollectionFilePath,
+                                            String clientCertChainFilePath,
+                                            String clientPrivateKeyFilePath) throws SSLException {
+    SslContextBuilder builder = GrpcSslContexts.forClient();
+    if (trustCertCollectionFilePath != null) {
+      builder.trustManager(new File(trustCertCollectionFilePath));
+    }
+    if (clientCertChainFilePath != null && clientPrivateKeyFilePath != null) {
+      builder.keyManager(new File(clientCertChainFilePath), new File(clientPrivateKeyFilePath));
+    }
+    return builder.build();
+  }
+
+  /**
+   * Construct client connecting to HelloWorld server at {@code host:port}.
+   */
+  public HelloWorldClientTls(String host,
+                             int port,
+                             SslContext sslContext) throws SSLException {
+
+    this(NettyChannelBuilder.forAddress(host, port)
+        .negotiationType(NegotiationType.TLS)
+        .sslContext(sslContext)
+        .build());
+  }
+
+  /**
+   * Construct client for accessing RouteGuide server using the existing channel.
+   */
+  HelloWorldClientTls(ManagedChannel channel) {
+    this.channel = channel;
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Say hello to server.
+   */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
```

Comment:
consider letting the exception propagate.

Reply:
this code i didn't write. it's copied from hello world existing